wordpress-mobile/WordPress-Android
# [Jetpack Content Migration Flow] Add Success Screen to the flow

Fixes #17373

> Depends on #17425 (which also has a dependency).
> The dependency PRs should be reviewed & merged into `trunk` first, afterwards the base branch of this PR should automatically change to `trunk`. At that point the `[Status] Not Ready for Merge` label can be removed, making this PR is ready for review.

This PR implements the last step of the WordPress to Jetpack migration flow:
- Adds UI for the Success Screen
- Adds navigation to the Success screen from the Notifications step
- Adds navigation to `My Site` from the Success screen

> 🏁 **All these changes are behind a temporary feature flag.**
>
> 🔔 **Notice:** Once the feature flag is enabled the migration flow will always be presented. The Jetpack app build has to be uninstalled (or "Clear Data" should be done) in order to revert back to the normal functionality. Further development on the data layer will make sure the migration is not attempted again once it's successful. To be partially handled in #17406.

To test:

0. Open the Jetpack app & login
1. Go to `Me` → `App Settings` → `Debug Settings`
2. Enable the `JetpackMigrationFlowFeatureConfig` & restart the app.
3. Tap the `Continue` button on the "Welcome to Jetpack!" screen
   - Note: Temporarily the migration progress is simulated by showing for 2 seconds a processing state.
4. Tap the `Continue` button on the "Notifications now come from Jetpack" screen
5. **Expect** to see the "Thanks for switching to Jetpack!" screen
6. Tap the `Finish` button
7. **Expect** to go back to the the "Welcome to Jetpack!" screen
   **Note** The functionality is implemented for navigating to the "My Site" screen, but since the feature flag is still enabled the migration flow is always shown instead of "My Site". See 🔔 notice above the testing steps.

## Previews

| Light | Dark |
| - | - |
| <img width="314" src="https://user-images.githubusercontent.com/4588074/200618446-4c73f8d5-0149-498b-ad8f-a981acc5389c.png"> | <img width="314" src="https://user-images.githubusercontent.com/4588074/200618458-1d280123-e916-4a63-9d3f-1d8982b93f79.png"> |

<details>
<summary>🎬 <b>Video Preview of the flow</b></summary>

Note: The video preview shows how the flow should behave. Currently the functionality from this PR brings us back to the Welcome step of the migration flow instead of the My Site screen.
This side effect of not landing on the "My Site" screen happens automatically because the logic for showing the migration flow depends on the development of the data layer which is still in WIP, and the temporary feature flag enables the migration flow forever on a test app build.

https://user-images.githubusercontent.com/4588074/200620301-861afe30-5718-48ec-bf26-bb495b735458.mp4

</details>

## Regression Notes

1. Potential unintended areas of impact
   None - the changes are to a new flow behind a feature flag.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
   N/a

3. What automated tests I added (or what prevented me from doing so)
   None - PR adds UI changes and trivial state changes.

PR submission checklist:
